editor/src/clj/editor/collection.clj

```diff
@@ -374,23 +372,41 @@
                                              (:resource (first source-build-targets)))))
 
 (g/defnk produce-proto-msg [name scale-along-z ref-inst-ddf embed-inst-ddf ref-coll-ddf]
-  {:name name
-   :scale-along-z (if scale-along-z 1 0)
-   :instances ref-inst-ddf
-   :embedded-instances embed-inst-ddf
-   :collection-instances ref-coll-ddf})
-
-(g/defnk produce-save-value [proto-msg]
-  (update proto-msg :embedded-instances
-          (fn [embedded-instance-descs]
-            (mapv (fn [embedded-instance-desc]
-                    (update embedded-instance-desc :data
-                            (fn [string-encoded-prototype-desc]
-                              (-> (protobuf/str->map GameObject$PrototypeDesc string-encoded-prototype-desc)
-                                  (game-object/strip-default-scale-from-components-in-prototype-desc)
-                                  (as-> prototype-desc
-                                        (protobuf/map->str GameObject$PrototypeDesc prototype-desc false))))))
-                  embedded-instance-descs))))
+  (protobuf/make-map-without-defaults GameObject$CollectionDesc
+    :name name
+    :scale-along-z (protobuf/boolean->int scale-along-z)
+    :instances ref-inst-ddf
+    :embedded-instances embed-inst-ddf
+    :collection-instances ref-coll-ddf))
+
+(defn instance-property-desc-save-value [instance-property-desc]
+  ;; GameObject$InstancePropertyDesc in map format.
+  (-> instance-property-desc
+      (protobuf/sanitize-repeated :properties game-object/component-property-desc-save-value)))
```

Comment:
* `protobuf/sanitize` works like `update`, but will exclude the key altogether if the `update-fn` returns `nil`.
* `protobuf/sanitize-repeated` works like `update (partial mapv update-fn)`, but will exclude the key altogether if the `update-fn` returns an empty sequence.

Elsewhere, you'll also see `protobuf/assign` and `protobuf/assign-repeated`, which are their `assoc` counterparts that will strip empty sequences or `nil` values in a similar fashion.